.github/workflows/build-service.yml

```diff
@@ -48,10 +65,16 @@ jobs:
 
   update-service-image:
     name: Update service image
-    needs: test-service
+    needs:
+      - matrix-gen
+      - test-service
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        stackTag: ${{ fromJSON(needs.matrix-gen.outputs.matrix) }}
     env:
       IMAGE_NAME: prompt-proto-service
+      STACK_TAG: ${{ matrix.stackTag }}
```

Comment:
Maybe call it BASE_TAG, to disambiguate the container label (e.g., `DM-37759-w_2023_06` or `latest`) from the actual Stack designation (`w_2023_06` or `d_latest`)?